portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -302,8 +302,7 @@ else if (PATH_PORTLET_FILE_ENTRY.equals(pathArray[0])) {
 
 					long imageCompanyId = image.getCompanyId();
 
-					if ((imageCompanyId != userCompanyId) &&
```

Comment:
The idea with this is that we've already checked the user's companyId, so there would be no need to check the image companyId if it's the same one as the user.